cycode/cli/utils/get_api_client.py

```
@@ -14,37 +14,58 @@
 
 
 def _get_cycode_client(
-    create_client_func: callable, client_id: Optional[str], client_secret: Optional[str], hide_response_log: bool
+    create_client_func: callable,
+    client_id: Optional[str],
+    client_secret: Optional[str],
+    hide_response_log: bool,
+    id_token: Optional[str] = None,
 ) -> Union['ScanClient', 'ReportClient']:
+    if id_token:
+        if not client_id:
+            raise click.ClickException('Cycode client id needed for OIDC authentication.')
+        return create_client_func(client_id, None, hide_response_log, id_token)
+
     if not client_id or not client_secret:
+        oidc_client_id, oidc_id_token = _get_configured_oidc_credentials()
+        if oidc_client_id and oidc_id_token:
```

Should we raise an error if we have an ID token configured, but no client ID?

---

Added this error
Also this logic has been restructured and will fallback to default approach if anything is missing